smarts/core/tests/test_collision.py

```diff
@@ -197,3 +206,51 @@ def test_ackerman_chassis_size_unchanged(bullet_client: bc.BulletClient):
     )
     collisions = step_with_vehicle_commands(chassis, steps=10)
     assert len(collisions) < 1
+
+
+AGENT_1 = "Agent_007"
+AGENT_2 = "Agent_008"
+
+
+@pytest.fixture
+def scenarios():
+    with temp_scenario(name="straight", map="maps/straight.net.xml") as scenario_root:
+        ego_missions = [
+            # missions of laner and buddha
+            t.Mission(t.Route(begin=("west", 0, 30), end=("east", 0, "max"),)),
+            t.Mission(t.Route(begin=("west", 0, 40), end=("east", 0, "max"),)),
+        ]
+        gen_scenario(
+            t.Scenario(ego_missions=ego_missions), output_dir=scenario_root,
+        )
+
+        yield Scenario.variations_for_all_scenario_roots(
+            [str(scenario_root)], [AGENT_1, AGENT_2]
+        )
+
+
+@pytest.fixture
+def smarts():
+    laner = AgentInterface(max_episode_steps=1000, action=ActionSpaceType.Lane,)
+    buddha = AgentInterface(max_episode_steps=1000, action=ActionSpaceType.Lane,)
+    agents = {AGENT_1: laner, AGENT_2: buddha}
+    smarts = SMARTS(
+        agents, traffic_sim=SumoTrafficSimulation(headless=True), envision=None,
+    )
+
+    yield smarts
+    smarts.destroy()
+
+
+def test_sim_level_collision(smarts, scenarios):
+    scenario = next(scenarios)
+    smarts.reset(scenario)
+
+    collisions = []
+
+    for _ in range(30):
+        observations, _, _, _ = smarts.step({AGENT_1: "keep_lane"})
```

Comment:
Nice test! I see that the updated test can catch the mistake in smarts.py, but doesn't catch the mistake in sensors.py. Perhaps it is more robust to check both events in observations and dones?

Reply:
Yes, good point. I updated the test.